src/main/scala/cromwell/engine/ExecutionStore.scala

```diff
@@ -0,0 +1,37 @@
+package cromwell.engine
+
+import cromwell.binding.{Call, WdlBinding}
+import cromwell.engine.ExecutionStatus.ExecutionStatus
+import scala.collection.mutable
+
+object ExecutionStatus extends Enumeration {
+  type ExecutionStatus = Value
+  val NotStarted, Running, Failed, Done = Value
+}
+
+/**
+ * Corresponds to the "execution table" of our discussions.
+ */
+class ExecutionStore(binding: WdlBinding) {
+
+  def isDone: Boolean = !table.exists(_._2 != ExecutionStatus.Done)
+
+  def updateStatus(call: Call, status: ExecutionStatus.Value): Unit = {
+    table.put(call, status)
+  }
+
+  private def buildInitialTable =
+    mutable.Map[Call, ExecutionStatus]() ++
+      binding.workflow.calls.map { call => call -> ExecutionStatus.NotStarted }.toMap
```

Comment:
I'm guessing this is one of those hacky corners that we'll need to address later (especially for scatter-gather).  This will definitely be addressed when we have the discussion about scoping.